shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Serialization.hs

```diff
@@ -56,31 +56,40 @@ import Cardano.Binary
     Size,
     ToCBOR (..),
     TokenType (TypeNull),
-    decodeBreakOr,
     decodeListLenOrIndef,
     decodeMapLenOrIndef,
     decodeNull,
     decodeTag,
-    decodeWord,
     encodeBreak,
     encodeListLen,
-    encodeListLenIndef,
     encodeMapLen,
     encodeMapLenIndef,
     encodeNull,
     encodeTag,
-    matchSize,
     peekTokenType,
     withWordSize,
   )
 import Cardano.Prelude (cborError)
-import Control.Monad (replicateM, unless)
+import Control.Monad (unless)
 import Data.Binary.Get (Get, getWord32le, runGetOrFail)
 import Data.Binary.Put (putWord32le, runPut)
 import qualified Data.ByteString as BS
 import qualified Data.ByteString.Builder as BS
 import qualified Data.ByteString.Builder.Extra as BS
 import qualified Data.ByteString.Lazy as BSL
+import Data.Coders
+  ( decodeCollection,
+    decodeCollectionWithLen,
+    decodeList,
+    decodeRecordNamed,
+    decodeRecordSum,
+    decodeSeq,
+    decodeSet,
+    decodeStrictSeq,
+    encodeFoldable,
+    encodeFoldableEncoder,
+    wrapCBORArray,
+  )
```

Comment:
If we're extracting these, maybe better to add them to cardano-binary (https://github.com/input-output-hk/cardano-base/tree/master/binary/src/Cardano/Binary). Check with @redxaxder